packages/docs/src/components/documentation.tsx

```diff
@@ -166,6 +169,26 @@ export class Documentation extends React.PureComponent<IDocumentationProps, IDoc
         Utils.safeInvoke(this.props.onComponentUpdate, activePageId);
     }
 
+    private getTitle() {
+        // Either get the title from the document, or return the default title
+        let title = "Blueprint - Documentation";
+        const docsTitle = document.getElementsByClassName("docs-title")[0].textContent;
+        // Check to make sure the textContent of docsTitle has a useable value
+        if (docsTitle !== undefined && docsTitle !== null) {
+            // Only change the title if we aren't on the index homepage
+            if (this.state.activePageId !== "blueprint" && this.state.activeSectionId !== "blueprint") {
```

Comment:
1. To be clear, the point of this helper is to avoid having a mutable variable at all (i.e. don't use `let`). Just `return` from this line directly.
1. Combine these `if` statements, since there's only one logic branch in here.
1. Also, since the `"Blueprint - Documentation"` case is the special one, let's put that in the `if` statement and make the other thing the default.

You can bundle most of these changes into much fewer lines in the end:

```tsx
const docsTitles = document.getElementsByClassName("docs-title");
const docsTitle = docsTitles.length > 0 ? docsTitles[0].textContent : undefined;

const isAtRootPage = this.state.activePageId === "blueprint" && this.state.activeSectionId === "blueprint";
return docsTitle == null || isAtRootPage ? "Blueprint - Documentation" : `${docsTitle} - Blueprint`;
```

Would that work as expected? ^

Reply:
Ah ok got it. Will run that, I believe that should work.